source/guide_prosody.rst

```diff
@@ -0,0 +1,309 @@
+
+#########
+Prosody
+#########
+
+Prosody_ is a modern XMPP communication server. It aims to be easy to set up and configure, and efficient with system resources. Additionally, for developers it aims to be easy to extend and give a flexible system on which to rapidly develop added functionality, or prototype new protocols.
+
+XMPP is an open and free alternative to commercial messaging and chat providers. Set it up for your company, organisation, or just your family and friends. You are in control, and your communication is private to you. Supporting a wide range of client software for desktop and mobile platforms, you can chat using Prosody from any device.
+
+Prosody is open source software under the permissive MIT/X11 license.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * supervisord_
+  * mysql_
+  * domains_
+  * ports_
+  * tls_
+
+Prerequisites
+=============
+Ports
+------------
+
+.. include:: includes/open-port.rst
+
+Altogether we need four open ports (referred to as: ``CLIENTPORT``, ``SERVERPORT`` and ``FILEUPLOADPORT`` ``FILETRANSFERPORT``) which should be public accessible.
+
+Domains, dns
+------------
+The following domains should be set up (to set up prosody for the domain ``stardust.space``):
+
+::
+ 
+ [isabell@stardust ~]$ uberspace web domain list
+ stardust.space
+ groupchat.stardust.space
+ [isabell@stardust ~]$
+
+Additionally to the dns_ A-records we need the following SRV-records: 
+
++---------------------------------------------+-------+-------+------+----------+--------+--------------+-----------------------+
+|  _service._proto.name			      | ttl   | class | type | priority | weight | port         | target                |
++=============================================+=======+=======+======+==========+========+==============+=======================+
+| _xmpp-client._tcp.stardust.space	      | 18000 | IN    | SRV  | 0        | 5      | CLIENTPORT   | stardust.uber.space   |
++---------------------------------------------+-------+-------+------+----------+--------+--------------+-----------------------+
+| _xmpp-server._tcp.stardust.space	      | 18000 | IN    | SRV  | 0        | 5      | SERVERPORT   | stardust.uber.space   |
++---------------------------------------------+-------+-------+------+----------+--------+--------------+-----------------------+
+| _xmpp-server._tcp.groupchat.stardust.space  | 18000 | IN    | SRV  | 0        | 5      | SERVERPORT   | stardust.uber.space   |
++---------------------------------------------+-------+-------+------+----------+--------+--------------+-----------------------+
+
+.. note:: Adapt the port-numbers and change ``stardust`` as well as ``stardust.space`` accordingly in your dns settings of your prefered provider.
+
+TLS
+---
+For simplicity we use openssl and the certificates created via letsencrypt_ for above domains which are already provided on the host after adding the domains and visiting them.
+
+Configure luarocks
```

Comment:
Hm, these environment variables should really be set automatically by us for all uberspaces. I'll create an issue internally, so we can skip this section here.

Reply:
Let me know what to change here.

Reply:
you can skip the entire "Configure luarocks" block :) we'll take care of that globally in the next release.